low-level/message_handlers/node_data_msg_handler.py

```diff
@@ -494,41 +513,47 @@ def _generate_host_update(self):
             else:
-                # Create the memory data message and hand it over
-                # to the egress processor to transmit
-                fault_resolved_event = "Host memory usage has decreased to {}%, "\
-                    "lower than the configured threshold of {}%.".format(
-                        self._node_sensor.total_memory["percent"],
-                        self._host_memory_usage_threshold
-                    )
-                logger.warning(fault_resolved_event)
-                logged_in_users = []
-
-                # Create the host update message and hand it over to the egress processor to transmit
-                hostUpdateMsg = HostUpdateMsg(self._node_sensor.host_id,
-                                        self._epoch_time,
-                                        self._node_sensor.boot_time,
-                                        self._node_sensor.up_time,
-                                        self._node_sensor.uname, self._units,
-                                        self._node_sensor.total_memory,
-                                        self._node_sensor.logged_in_users,
-                                        self._node_sensor.process_count,
-                                        self._node_sensor.running_process_count,
-                                        self.FAULT_RESOLVED,
-                                        fault_resolved_event
-                                        )
-
-                # Add in uuid if it was present in the json request
-                if self._uuid is not None:
-                    hostUpdateMsg.set_uuid(self._uuid)
-                jsonMsg = hostUpdateMsg.getJson()
-                # Transmit it to message processor
-                self.host_sensor_data = jsonMsg
-                self.os_sensor_type["memory_usage"] = self.host_sensor_data
-                self._write_internal_msgQ(EgressProcessor.name(), jsonMsg)
-                self.high_usage['memory'] = False
-                self.usage_time_map['memory'] = int(-1)
-                self.persist_state_data('memory', 'MEMORY_USAGE_DATA')
+                if fault_resolved_iters < iteration_limit:
+                    fault_resolved_iters += 1
+                    self.fault_resolved_iterations['memory'] = fault_resolved_iters
+                    self.persist_state_data('memory', 'MEMORY_USAGE_DATA')
+                elif fault_resolved_iters >= iteration_limit:
+                    # Create the memory data message and hand it over
+                    # to the egress processor to transmit
+                    fault_resolved_event = "Host memory usage has decreased to {}%, "\
+                        "lower than the configured threshold of {}%.".format(
+                            self._node_sensor.total_memory["percent"],
+                            self._host_memory_usage_threshold
+                        )
+                    logger.warning(fault_resolved_event)

```

Comment:
Why warning, its a good event, can be logger.info !
I see this is how it was from earlier

Reply:
Done.

Reply:
Updated for disk fault resolved event as well.